### PR TITLE
(FM-6054) - Unit tests for low effort functions

### PR DESCRIPTION
**Pull request by pmcmaw**
This is unit tests for the low effort function tests for i18n compatibility. 

This includes tests for the following functions:
delete_undef_values
difference
flatten
fqdn_rand_string
fqdn_rotate
grep
hash
intersection
join
join_keys_to_values
lstrip
member
merge
pick_default
prefix
reject
reverse
rstrip
strip
suffix
union
unique

### CODE REVIEW

File: spec/functions/join_keys_to_values_spec.rb [hunk 1]
```diff
@@ -10,7 +10,12 @@
   it { is_expected.to run.with_params({}, '').and_return([]) }
   it { is_expected.to run.with_params({}, ':').and_return([]) }
   it { is_expected.to run.with_params({ 'key' => 'value' }, '').and_return(['keyvalue']) }
+  it { is_expected.to run.with_params({ 'ҝẽγ' => '√ạĺűē' }, '').and_return(['ҝẽγ√ạĺűē']) }
   it { is_expected.to run.with_params({ 'key' => 'value' }, ':').and_return(['key:value']) }
+  it { is_expected.to run.with_params({ 'ҝẽγ' => '√ạĺűē' }, ':').and_return(['ҝẽγ:√ạĺűē']) }
+  it { is_expected.to run.with_params({ 'ҝẽγ' => '√ạĺűē' }, '万').and_return(['ҝẽγ万√ạĺűē']) }
+
```
Comment (tphoney) [inline spec/functions/join_keys_to_values_spec.rb]: i cant see the difference between these added tests, why 3 of them ?

Comment (tphoney) [inline spec/functions/join_keys_to_values_spec.rb]: otherwise looks good

Comment (pmcmaw) [inline spec/functions/join_keys_to_values_spec.rb]: The first test is testing the actual functionality of the function, the second test is making sure that the function can cope with UTF8 strings, then the final test is ensuring that the double byte character can be used.

Comment (pmcmaw) [inline spec/functions/join_keys_to_values_spec.rb]: I have now updated the context for the above tests to make it more clear what the test is doing and its purpose.

**Pull request merged**